source/MRMesh/MRPolylineTrimWithPlane.h

```diff
@@ -1,22 +1,27 @@
 #pragma once
-#pragma once
 #include "MRMeshFwd.h"
+#include "MRBitSet.h"
 
 namespace MR
 {
 /// This function splits edges intersected by the plane
-/// \return New edges with origin on the plane and oriented to the positive direction
+/// \return edges located above the plane (in direction of normal to plane)
 /// \param polyline Input polyline that will be cut by the plane
 /// \param plane Input plane to cut polyline with
+/// \param newPositiveEdges new edges with origin on the plane and oriented to the positive direction
 /// \param onEdgeSplitCallback is invoked each time when an edge is split. Receives edge ID before split, edge ID after split, and weight of the origin vertex
-MRMESH_API EdgeBitSet subdividePolylineWithPlane( Polyline3& polyline, const Plane3f& plane, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback = nullptr );
+MRMESH_API EdgeBitSet subdivideWithPlane( Polyline3& polyline, const Plane3f& plane, EdgeBitSet* newPositiveEdges = nullptr, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback = nullptr );
+[[deprecated]] inline EdgeBitSet subdividePolylineWithPlane( Polyline3& polyline, const Plane3f& plane, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback = nullptr )
```

Comment:
If it requires MRBitSet.h include, please put it inside .cpp

Reply:
fixed